cmd/security-secrets-setup/main.go

```diff
@@ -16,80 +17,98 @@ package main
 
 import (
 	"flag"
-	"log"
+	"fmt"
 	"os"
 	"time"
 
 	"github.com/edgexfoundry/edgex-go/internal"
-	"github.com/edgexfoundry/edgex-go/internal/pkg/config"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/usage"
 	"github.com/edgexfoundry/edgex-go/internal/security/setup"
-	"github.com/edgexfoundry/edgex-go/internal/security/setup/certificates"
-	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/edgexfoundry/edgex-go/internal/security/setup/option"
 )
 
-func main() {
-	start := time.Now()
-	var configFile string
+type exit interface {
+	callExit(int)
+}
+
+type exitCode struct{}
+
+type optionDispatcher interface {
+	run() (int, error)
+}
+
+type pkiInitOptionDispatcher struct{}
+
+var exitInstance = newExit()
+var dispatcherInstance = newOptionDispatcher()
+var configFile string
 
+func init() {
+	// define and register command line flags:
 	flag.StringVar(&configFile, "config", "", "specify JSON configuration file: /path/to/file.json")
 	flag.StringVar(&configFile, "c", "", "specify JSON configuration file: /path/to/file.json")
 	flag.Usage = usage.HelpCallbackSecuritySetup
-	flag.Parse()
-
-	if configFile == "" {
-		flag.PrintDefaults()
-		os.Exit(1)
-	}
+}
 
-	setup.Init()
+func main() {
+	start := time.Now()
 
-	// Create and initialize the fs environment and global vars for the PKI materials
-	lc := logger.NewClient("security-secrets-setup", setup.Configuration.Logging.EnableRemote,
-		setup.Configuration.Logging.File, setup.Configuration.Writable.LogLevel)
+	flag.Parse()
 
-	// Read the Json config file and unmarshall content into struct type X509Config
-	x509config, err := config.NewX509Config(configFile)
-	if err != nil {
-		lc.Error(err.Error())
+	if len(os.Args) < 2 {
+		fmt.Println("Please specify option for " + option.SecuritySecretsSetup)
+		flag.PrintDefaults()
+		exitInstance.callExit(1)
```

Comment:
@anonymouse64 Ian, I am not against the exit code 0 for --help or -h option.  So it is a matter how we treat no option is provided at all. Currently we treat no option provided as exit code 1 instead of 0 but I am open to that.  @bnevis-i  what do you think?

Reply:
I have seen Unix utilities that do it both ways.  Personally I think if the command didn't do its intended job it should return a non-zero exit status, so that thinks like exec.Command will flag the error.  But since there is no generally accepted standard, if Ian says zero status when displaying usage is the EdgeX convention, I can go along with that.

Reply:
ok, i am ok to go with 0 status code then